source/users/enable-custom-user-data.txt

```diff
@@ -72,6 +72,54 @@ configure custom user data for your application, you specify:
 - the user ID field used to map custom user data documents
   to users (via user ID)
 
+- an optional function that runs before new user accounts are created
+
+.. _user-creation-function:
+
+User Creation Function
+~~~~~~~~~~~~~~~~~~~~~~
+
+You can define a function that runs every time a new user successfully
+registers but before their new user account is created. If the function
+throws or otherwise errors, the user account creation fails. This lets
+you ensure that users always have custom data associated with them once
+created.
+
+The function receives a user metadata object as its only argument. You
```

Comment:
can you link to a ref for 'user metadata object'? prob ref at top of this page https://www.mongodb.com/docs/atlas/app-services/authentication/user-objects/